exporter/loadbalancingexporter/exporter.go

```diff
@@ -16,42 +16,208 @@ package loadbalancingexporter
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
 
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configmodels"
 	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/otlpexporter"
 	"go.uber.org/zap"
 )
 
 var _ component.TracesExporter = (*exporterImp)(nil)
 
+const (
+	defaultPort = "55680"
+)
+
+var errNoResolver = errors.New("no resolvers specified for the exporter")
+
 type exporterImp struct {
 	logger *zap.Logger
 	config Config
+
+	res  resolver
+	ring *hashRing
+
+	exporters            map[string]component.TracesExporter
+	exporterFactory      component.ExporterFactory
+	templateCreateParams component.ExporterCreateParams
+
+	stopped    bool
+	shutdownWg sync.WaitGroup
+	updateLock sync.RWMutex
 }
 
 // Crete new exporter
 func newExporter(params component.ExporterCreateParams, cfg configmodels.Exporter) (*exporterImp, error) {
 	oCfg := cfg.(*Config)
 
+	tmplParams := component.ExporterCreateParams{
+		Logger:               params.Logger,
+		ApplicationStartInfo: params.ApplicationStartInfo,
+	}
+
+	var res resolver
+	if oCfg.Resolver.Static != nil {
+		var err error
+		res, err = newStaticResolver(oCfg.Resolver.Static.Hostnames)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if oCfg.Resolver.DNS != nil { // if both are configured, DNS takes precedence
```

Comment:
Should this be considered a configuration error? Not sure

Reply:
Good question. I typically prefer to be lenient with what I get, but this might indeed be a case where we want to be very clear and not guess what the user might have wanted.

Reply:
I added a check before this, so that only one of the two conditions will now be true if this code is reached.